lib/pbench/test/unit/server/test_requests.py

```diff
@@ -59,25 +58,28 @@ class FakeTarball:
             def __init__(self, path: Path):
                 self.tarball_path = path
                 self.name = Dataset.stem(path)
+                self.metadata = None
 
             def delete(self):
                 TestUpload.tarball_deleted = self.name
 
             def get_metadata(self):
-                return {"pbench": {"date": "2002-05-16T00:00:00"}}
+                self.metadata = {"pbench": {"date": "2002-05-16T00:00:00"}}
 
```

Comment:
Whole Cache manager was a fake. So, according to the way it was organized it was behaving correctly because I did not pass the metadata and assigned it directly.